components/kyma-environment-broker/internal/process/provisioning/cls_binding_step.go

```diff
@@ -0,0 +1,174 @@
+package provisioning
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/cls"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/runtime/components"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/servicemanager"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockery --name=ClsBindingProvider --output=automock --outpkg=automock --case=underscore
+type ClsBindingProvider interface {
+	CreateBinding(smClient servicemanager.Client, request *cls.BindingRequest) (*cls.ClsOverrides, error)
+}
+
+type ClsBindStep struct {
+	config           *cls.Config
+	operationManager *process.ProvisionOperationManager
+	secretKey        string
+	bindingProvider  ClsBindingProvider
+}
+
+func NewClsBindStep(config *cls.Config, bp ClsBindingProvider, os storage.Operations, secretKey string) *ClsBindStep {
+	return &ClsBindStep{
+		config:           config,
+		operationManager: process.NewProvisionOperationManager(os),
+		secretKey:        secretKey,
+		bindingProvider:  bp,
+	}
+}
+
+var _ Step = (*ClsBindStep)(nil)
+
+func (s *ClsBindStep) Name() string {
+	return "CLS_Bind"
+}
+
+func (s *ClsBindStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if !operation.Cls.Instance.ProvisioningTriggered {
+		failureReason := fmt.Sprintf("cls provisioning step was not triggered")
+		log.Errorf("%s: %s", failureReason, "")
+		return s.operationManager.OperationFailed(operation, failureReason)
+	}
+
+	skrRegion := operation.ProvisioningParameters.Parameters.Region
+	smRegion, err := cls.DetermineServiceManagerRegion(skrRegion)
+	smCredentials, err := cls.FindCredentials(s.config.ServiceManager, smRegion)
+	smCli := operation.SMClientFactory.ForCredentials(smCredentials)
```

Comment:
The CLS region is already stored in the CLS section of the operation. Inferring it from the SKR region in this case would be wrong
```suggestion
smCredentials, err := cls.FindCredentials(s.config.ServiceManager, operation.Cls.Region)
	if err != nil {
		failureReason := fmt.Sprintf("Unable to find credentials for cls service manager in region %s: %s", operation.Cls.Region, err)
		log.Error(failureReason)
		return s.operationManager.OperationFailed(operation, failureReason)
	}

	smClient := operation.SMClientFactory.ForCredentials(smCredentials)
```

Reply:
done